### PR TITLE
feat(components): [Carousel]  Add motion blur functionality

### PR DESCRIPTION
**Pull request by JedediahXu**
* docs(components): [Carousel] Add example properties

* docs(components): [Carousel] Add docs properties

Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

# Why add motion blur  为什么要添加运动模糊

Add motion blur for a better user experience 

添加运动模糊以获得更好的用户体验

## 特性

1、Smooth Transition Effect: Motion blur can create a smooth transition effect in the carousel, making the images appear more natural during switches and reducing the abruptness associated with sudden transitions.
     平滑过渡效果: 运动模糊可以在轮播中创建一个平滑的过渡效果，使图像在切换时看起来更自然，减少突然过渡带来的突兀感。

2、Highlighting Focus: Emphasizing the primary content within the current carousel slide, making it more prominent during transitions.
    突出焦点: 强调当前轮播幻灯片中的主要内容，使其在过渡期间更加突出。

3、Dynamics and Fluidity: Introducing a sense of dynamism, providing users with a smoother and captivating visual experience, thereby enhancing overall appeal.
     动态和流体感: 增加了动态感，为用户提供了更流畅和引人入胜的视觉体验，从而增强整体吸引力。

4、Modernity and Artistry: Infused a touch of modernity and artistic flair to the carousel, giving the page a creative and stylish appearance.
     现代感和艺术感: 为轮播增添了一丝现代感和艺术气息，让页面具有创意和时尚的外观。

5、Reducing Eye Fatigue: Smooth motion effects help alleviate potential eye strain for users when viewing carousels.
     减少眼睛疲劳: 平滑的运动效果有助于减少用户在观看轮播时可能产生的眼睛疲劳 

Additionally, it seems that this feature is not yet available in other Vue component libraries.
此外，这个特性在其他Vue组件库中似乎还不可用。

##  Any Break Changes 是否有破坏性

No destructive update
无破坏性更新

## discuss 讨论

#15399

### CODE REVIEW

**Comment by btea** [comment]
The test you added seems to be failing, please check.

**Comment by JedediahXu** [comment]
> The test you added seems to be failing, please check.

receive

File: packages/components/carousel/src/carousel.ts [hunk 1]
```diff
@@ -51,6 +51,10 @@ export const carouselProps = buildProps({
     type: Boolean,
     default: true,
   },
+  motionBlur: {
+    type: Boolean,
+    default: false,
```
Comment (btea) [inline packages/components/carousel/src/carousel.ts:56]: The default value here can be removed.

File: packages/components/carousel/src/carousel.vue [hunk 2]
```diff
@@ -111,6 +132,16 @@ const carouselClasses = computed(() => {
   return classes
 })
 
+const carouselContainer = computed(() => {
+  const classes = [ns.e('container')]
+  if (props.motionBlur && unref(isTransitioning)) {
+    classes.push(
+      unref(isVertical) ? 'el-transitioning-vertical' : 'el-transitioning'
```
Comment (btea) [inline packages/components/carousel/src/carousel.vue:139]:
```suggestion
      unref(isVertical) ? `${ns.namespace}-transitioning-vertical`: `${ns.namespace}-transitioning`
```

File: packages/theme-chalk/src/carousel.scss [hunk 3]
```diff
@@ -178,3 +178,11 @@
   transform: translateY(-50%) translateX(10px);
   opacity: 0;
 }
+
+.el-transitioning {
```
Comment (btea) [inline packages/theme-chalk/src/carousel.scss:182]:
```suggestion
.#{$namespace}-transitioning {
```

File: packages/theme-chalk/src/carousel.scss [hunk 4]
```diff
@@ -178,3 +178,11 @@
...
+  filter: url('#horizontal');
+}
+
+.el-transitioning-vertical {
```
Comment (btea) [inline packages/theme-chalk/src/carousel.scss:186]:
```suggestion
.#{$namespace}-transitioning-vertical {
```

File: packages/components/carousel/src/carousel.ts (hunk 1 above)
Comment (JedediahXu) [reply packages/components/carousel/src/carousel.ts]: > The default value here can be removed.

receive

**Pull request closed without merging**